### DISCUSSION TITLE
hub.docker.com sallius/tsun-gen3-proxy:latest (vom 25.05.2025) - funktioniert nicht korrekt

### DISCUSSION THREAD

**Discussion by ryanrudak** [s-allius/tsun-gen3-proxy#76] [General]
Moin, 

habe mir das letzte Docker Image mit dem Digests sha256:8260cdbbcd52fe51a6d9c2edb02b1a74d928778aa4e4ea52abfaf47147180a71 vom 25.05.2024 mal angesehen. Irgendwie funktioniert dies nicht korrekt. Man bekommt nicht mehr alle Geräte und die dazugehörigen Entitäten ins HASSIO rein. Ich habe einen MS1600 und bekomme nur den Controller, den Micros Inverter, das Module PV1 und den Proxy als Geräte. PV2, PV3 und PV4 werden als Geräte nicht mehr erstellt. Im Logfile ist leider nicht wirklich was zu erkennen, bin allerdings auch nicht im DEBUG Mode gewesen. Die Daten für die Cloud waren aber vollständig und aktuell.

Ich bin dann auf das Docker Image mit dem Digests 7b2c7bc96a0e39e9e807488ea04f0de76148f8dca81c603994878529244593bf von vor 4 Tages zurückgegangen und damit funktioniert alles.

Zudem ist mir aufgefallen, das wenn der Micro Inverter nicht mehr erreichbar ist, wird das Logfile sehr schnell, sehr voll geschrieben und hat in kurzer Zeit schon mal knapp 10 GB (also das Log von Docker Container und das proxy.log) geschrieben, da ist die Platte meiner VM schnell voll und das System steht sozusagen.

**Comment by s-allius** [comment]
Danke fürs Feedback. Ich schaue mir das morgen mal an, was da schief gelaufen ist.

Die Größe der Log ist mir auch schon aufgefallen. Bisher wollte ich immer aller Pakete an den WR loggen. Mit den Modbus Paketen wird das aber einfach zu viel, insbesondere da ich jetzt auch noch eine Wiederholung eingebaut habe...

**Reply by s-allius** [reply]
Für die Reduzierung der log Dateien habe ich ein Ticket #77  aufgemacht

**Comment by s-allius** [comment]
Hmm ich komme da nicht hinter. Die sha hashes gehören zu deinen Docker Container, die du selbst gebaut hast, richtig? In dem Zeitraum habe ich nur Änderungen bezüglich der Logs eingecheckt. Also mit einem Blick auf den Code sehe ich da nichts.
Hast du mit dem letzten Image crashes im Log?
Das Problem ist doch, dass wenn du die Entitäten in HA löschst, dass sie nicht wieder angelegt werden? Oder fehlen nur Messwerte?

**Comment by ryanrudak** [comment]
Ich meine schon das Docker Image von hub.docker.com, weil das Erstellen eines lokalen Images immer Fehler gebracht hat bei den 'hardening scripts'.

![image](https://github.com/s-allius/tsun-gen3-proxy/assets/51046918/c653b48a-027a-4453-bf56-9a3edd6024d9)

Genau, ich habe in HASSIO alle Entitäten gelöscht und PV2, PV3 und PV4 wurden nicht mehr erstellt.

Ich habe aber heute noch mal auf meinem Linux Host alles von Docker runtergeschmissen und nach Anleitung von docs.docker.com installiert. Keine Ahnung was nun anders ist, aber ich konnte ein lokales Image mittels "docker build https://github.com/s-allius/tsun-gen3-proxy.git#main:app -t tsun-proxy" erstellen und der daraus gestartete Container läuft und aktualisiert auch alle Werte (soweit ich das in den letzten 20 Minuten gesehen habe).

Vielen Dank für die tolle Arbeit!

**Reply by s-allius** [reply]
Ok. Das Problem ist vermutlich, dass aus irgendeinen Grund die Anzahl der MPTT nicht richtig ermittelt wurde. Wenn diese Info fehlt oder zu spät kommt, dann wird nur ein Eingang registriert. Das fällt vermutlich kaum auf, da es nur einmal funktionieren muss.
Ich werde mal darauf achten, ob ich da noch einen Fehler finde oder etwas verbessern kann.


Auf docker.io habe ich immer die aktuellen Entwickler Images liegen. Da kann auch schon mal eins defekt sein, da ich darüber die Images teste bevor ich die Änderungen einchecke. Damit sind die Änderungen aber auch noch nicht im GIT und eine Fehlversuch macht auch kein Sinn.


Wenn die hardening Skripte nicht laufen, würde ich das gerne fixen. Das liegt vermutlich an der Entwicklungsumgebung bei dir. Evtl. fehlt da was oder eine version ist nicht kompatibel. Wenn du einen Trace mal hast, dann gerne nochmal ein ticket öffnen
